### PR TITLE
Add com.inklestudios.Inky

### PR DESCRIPTION
**Pull request by manuq**
Add Inkle Studio's [Inky IDE](https://www.inklestudios.com/ink/). It's an Electron app licensed under the MIT license. The upstream repository is: https://github.com/inkle/inky/

Although Inky is in alpha, it works very well for me. At Endless we are using it to write branching narrative in games.

Since the [Linux build uploaded to upstream](https://github.com/inkle/inky/releases) is outdated and doesn't work, I took the liberty to upload a ZIP of a working build [to my cloned repository](https://github.com/manuq/inky/releases/tag/alpha). That is where the archive source points in the manifest.

The desktop file is inspired by [the one for Appimage](https://github.com/inkle/inky/blob/master/resources/com.inkle.inky.desktop)

The icon file is a 512 x 512 rescaled version of [the one for AppImage](https://github.com/inkle/inky/blob/master/resources/Icon1024.png).

### CODE REVIEW

**Comment by manuq** [comment]
bot, build com.inklestudios.Inky

**Comment by manuq** [comment]
Added a `flathub.json` because this is only for architecture x86_64.

**Comment by manuq** [comment]
bot, build com.inklestudios.Inky

File: com.inklestudios.Inky.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                {
+                    "type": "script",
+                    "dest-filename": "inky-wrapper.sh",
+                    "commands": [
+                        "exec /app/inky/Inky \"$@\""
+                    ]
+                },
+                {
+                    "type": "file",
+                    "path": "com.inklestudios.Inky.appdata.xml"
```
Comment (LeandroStanger) [inline com.inklestudios.Inky.json:54]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

File: com.inklestudios.Inky.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+  <id>com.inklestudios.Inky.desktop</id>
```
Comment (bilelmoussaoui) [inline com.inklestudios.Inky.appdata.xml:3]:
```suggestion
  <id>com.inklestudios.Inky</id>
  <launchable type="desktop-id">com.inklestudios.Inky.desktop</launchable>
```

File: com.inklestudios.Inky.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  </description>
+  <screenshots>
+    <screenshot type="default">
+      <image type="source">https://www.inklestudios.com/ink/img/inky-screenshot@2x.png</image>
+    </screenshot>
+  </screenshots>
+  <categories>
+    <category>IDE</category>
+    <category>Development</category>
+  </categories>
```
Comment (bilelmoussaoui) [inline com.inklestudios.Inky.appdata.xml:27]:
```suggestion
```
The categories are merged from the desktop file, you don't have to repeat that in the metainfo file

File: com.inklestudios.Inky.desktop [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env xdg-open
```
Comment (bilelmoussaoui) [inline com.inklestudios.Inky.desktop:1]:
```suggestion
```

**Comment by bilelmoussaoui** [comment]
Have you reached out to the upstream maintainers if they were interested in (co)-maintaining their app on Flathub? That would be a great opportunity to maybe ask them for a new tarball for Linux :)

File: com.inklestudios.Inky.json (hunk 1 above)
Comment (manuq) [reply com.inklestudios.Inky.json:54]: Thanks for your review. You mean the file suffix should be `.metainfo.xml`? Or is there anything to change in the format?

Comment (lu9dce) [reply com.inklestudios.Inky.json:54]: .metainfo.xml

**Comment by manuq** [comment]
> Have you reached out to the upstream maintainers if they were interested in (co)-maintaining their app on Flathub? That would be a great opportunity to maybe ask them for a new tarball for Linux :)

Upstream seems very busy with their new game :) Hopefully they fire a new release soon. I just asked in the following issue: https://github.com/inkle/inky/issues/292#issuecomment-737544782

File: com.inklestudios.Inky.json (hunk 1 above)
Comment (manuq) [reply com.inklestudios.Inky.json]: Thanks for clarifyng. I just pushed the rename.

**Comment by manuq** [comment]
bot, build com.inklestudios.Inky

File: com.inklestudios.Inky.desktop [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
```
Comment (barthalion) [inline com.inklestudios.Inky.desktop:1]: Ideally this, appdata and icon files should be upstreamed.

**Comment by barthalion** [comment]
@manuq Anyone from Endless beside you who should be given write access?

File: com.inklestudios.Inky.desktop (hunk 5 above)
Comment (manuq) [reply com.inklestudios.Inky.desktop:1]: Indeed. I will open a PR for them.

Comment (manuq) [reply com.inklestudios.Inky.desktop:1]: @barthalion this is the desktop file they have upstream: https://github.com/inkle/inky/blob/master/resources/com.inkle.inky.desktop . It is for AppImage. Do you think there is a way to merge this one without affecting AppImage?

1. Is the shebang `#!/usr/bin/env xdg-open` mandatory in appimage?
2. Maybe I can use `desktop-file-edit` in the manifest to change `Exec` and `Path`?
3. It seems they have the app ID wrong, the studio's domain is clearly `inklestudios.com`, not `inkle.com`.

**Comment by manuq** [comment]
> @manuq Anyone from Endless beside you who should be given write access?

Please also give write access to @danigm . Thanks.

File: com.inklestudios.Inky.desktop (hunk 5 above)
Comment (barthalion) [reply com.inklestudios.Inky.desktop:1]: 1. No. It's probably a hack to run a desktop file from CLI after `chmod +x`.
2. Yeah, that's what maintainers usually do.
3. Apparently so! Feel free to fix that after application is merged, it's not a blocker.

**Pull request closed without merging**